### PR TITLE
views: filter records and facets by permissions

### PR DESCRIPTION
**Pull request by ammirate**
This commit introduces a permission check for literature and jobs endpoints.

* ADD new Enum class calles Roles centralize the roles in Inspire;
* ADD new faces 'status' for job, if user is cataloger;
* REFACTOR all the superuser and cataloger permissions using Roles;

* INSPIR-2409

Signed-off-by: Antonio Cesarano <cesarano2607@gmail.com>

### CODE REVIEW

File: inspirehep/accounts/__init__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from enum import Enum
+
+
+class Roles(Enum):
```
Comment (drjova) [inline inspirehep/accounts/__init__.py]: move this to ``constants.py`` WDYT?

File: tests/integration/records/views/test_views_jobs/es_aggregations_for_1735925.json [hunk 2]
```diff
@@ -22,3 +22,4 @@
     "buckets": [{ "key": "North America", "doc_count": 1 }]
   }
 }
+
```
Comment (drjova) [inline tests/integration/records/views/test_views_jobs/es_aggregations_for_1735925.json]: oups

File: tests/integration/records/views/test_views_literature.py [hunk 3]
```diff
@@ -542,3 +547,96 @@ def test_literature_citation_annual_summary_for_many_records(
...
+    response_status_code = response.status_code
+    response_data = json.loads(response.data)
+    assert response_data["hits"]["total"] == 1
+
+    response_data_metadata = response_data["hits"]["hits"][0]["metadata"]
+
+    assert expected_status_code == response_status_code
+    assert expected_data == response_data_metadata
+
+    print(api_client.session_transaction)
```
Comment (drjova) [inline tests/integration/records/views/test_views_literature.py]: left over

File: tests/integration/records/views/test_views_literature.py [hunk 4]
```diff
@@ -542,3 +547,96 @@ def test_literature_citation_annual_summary_for_many_records(
...
+        current_user.authenticated = False
+        current_user.roles = []
+
+    create_record("lit", data={"_collections": ["Fermilab"]})
+    create_record("lit")
+
+    headers = {"Accept": "application/json"}
+    response = api_client.get("/literature", headers=headers)
+    response_data = json.loads(response.data)
+    assert response_data["hits"]["total"] == 1
```
Comment (drjova) [inline tests/integration/records/views/test_views_literature.py]: shall we check the control number here? Just to make sure the response is the ```_collections: ["Literature"]``` etc

File: tests/integration/records/views/test_views_literature.py [hunk 5]
```diff
@@ -542,3 +547,96 @@ def test_literature_citation_annual_summary_for_many_records(
...
+        "$schema": "http://localhost:5000/schemas/records/hep.json",
+        "_collections": ["Fermilab"],
+        "control_number": 666,
+        "document_type": ["article"],
+        "titles": [{"title": "Partner walk again seek job."}],
+    }
+
+    create_record("lit", data=data)
+
+    headers = {"Accept": "application/json"}
```
Comment (drjova) [inline tests/integration/records/views/test_views_literature.py]: you can remove headers by default is ``application/json``

File: inspirehep/search/utils.py [hunk 6]
```diff
@@ -9,14 +9,30 @@
...
+    facet_name = request.values.get("facet_name")
 
-    if not facet:
-        facet = current_app.config["RECORDS_REST_FACETS"].get(search_index)
+    if is_superuser_or_cataloger_logged_in():
+        cataloger_facet = current_app.config["CATALOGER_RECORDS_REST_FACETS"]
+        facet = cataloger_facet.get(facet_name) or cataloger_facet.get(search_index)
+    else:
+        user_facet = current_app.config["RECORDS_REST_FACETS"]
+        facet = user_facet.get(facet_name) or user_facet.get(search_index)
```
Comment (drjova) [inline inspirehep/search/utils.py]:
```python
if is_superuser_or_cataloger_logged_in():
   facet_data = 
else:
   facet_data = 
facet_data.get(facet_name) or facet_data.get(search_index)
```
Just to have once the logic of getting ```facet_name``` or ``search_index``

File: inspirehep/accounts/__init__.py (hunk 1 above)
Comment (ammirate) [reply inspirehep/accounts/__init__.py]: I'll go for `roles.py` :+1:

File: inspirehep/accounts/__init__.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    superuser = "superuser"
+    cataloger = "cataloger"
+    hermescoll = "hermescoll"
+    hermescurator = "hermescurator"
+    jlabcurator = "jlabcurator"
```
Comment (michamos) [inline inspirehep/accounts/__init__.py]: this thing should really be an enum at the DB level, unfortunately we can't do that :cry:

File: inspirehep/search/api.py [hunk 8]
```diff
@@ -98,14 +98,17 @@ class LiteratureSearch(InspireSearch):
...
+        default_filter = DefaultFilter(Q())
 
     def query_from_iq(self, query_string):
         """Initialize ES DSL object using INSPIRE query parser.
         :param query_string: Query string as a user would input in INSPIRE's search box.
         :type query_string: string
         :returns: Elasticsearch DSL search class
         """
+        if not is_superuser_or_cataloger_logged_in():
+            user_query = Q(IQ(query_string, self) & Q("match", _collections="Literature"))
```
Comment (michamos) [inline inspirehep/search/api.py]: it's better to use filters for this kind of stuff, they are faster as they don't influence the score and can be cached (see [here](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/query-filter-context.html) for more details).

File: inspirehep/search/api.py [hunk 9]
```diff
@@ -181,6 +184,17 @@ class Meta:
...
         doc_types = "jobs"
 
+    def query_from_iq(self, query_string):
+        """Initialize ES DSL object using INSPIRE query parser.
+        :param query_string: Query string as a user would input in INSPIRE's search box.
+        :type query_string: string
+        :returns: Elasticsearch DSL search class
+        """
+        if not is_superuser_or_cataloger_logged_in():
+            user_query = Q(IQ(query_string, self) & Q("match", status="open"))
```
Comment (michamos) [inline inspirehep/search/api.py]: same

File: tests/integration/conftest.py [hunk 10]
```diff
@@ -243,3 +243,21 @@ def _create_user(role="user", orcid=None):
         return UserFactory(role=role, orcid=orcid)
 
     return _create_user
+
+
+@pytest.fixture
+def logout():
```
Comment (drjova) [inline tests/integration/conftest.py]: ❤️

File: tests/integration/records/views/test_views_jobs.py [hunk 11]
```diff
@@ -162,3 +170,90 @@ def test_jobs_accelerator_experiments(api_client, db, es_clear, create_record, d
...
+    assert expected_results == result.json["hits"]["total"]
+
+
+def test_jobs_search_permissions(
+    api_client, db, es_clear, create_record, datadir, create_user, logout
+):
+    create_record("job", data={"status": "pending"})
+    create_record("job", data={"status": "open"})
+
+    headers = {"Accept": "application/json"}
```
Comment (drjova) [inline tests/integration/records/views/test_views_jobs.py]: leftovers?

**Pull request merged**